openedx/frontend-app-learning
# Added Courseware Search container popover

## Description
Ticket: [KBK-42 🔒](https://2u-internal.atlassian.net/browse/KBK-42)

We need a [modal 🔒](https://2u-internal.atlassian.net/wiki/spaces/Kubrick/pages/614432780/Component+breakdown+based+on+Figma#Full-Screen-Modal) to show the search components and the results.

> [!IMPORTANT]
> I've added `@testing-library/react-hooks@^8.0.1` dependency. This allows testing standalone hooks rather than creating a testing component to use the hook. It makes it much cleaner to test. 

## UI changes

### instructor view

| 480px wide  | 1024px wide |
| --- | --- |
| ![search-w480-instructor](https://github.com/openedx/frontend-app-learning/assets/12736783/d54e7b49-de83-463b-98e8-8f43c81fd506) | ![search-w1024-instructor](https://github.com/openedx/frontend-app-learning/assets/12736783/149af82a-848f-4418-b077-84bec2974f58) |

### Student view

| 480px wide  | 1024px wide |
| --- | --- |
| ![search-w480-student](https://github.com/openedx/frontend-app-learning/assets/12736783/27715bbf-90e0-420c-8ab2-5245dfd685c0) | ![search-w1024-student](https://github.com/openedx/frontend-app-learning/assets/12736783/14a1d57e-e423-4ea6-99ed-07740a1c9a45) |

The Design is a bit peculiar, and I couldn't match the UI using Paragon's [FullscreenModal](https://paragon-openedx.netlify.app/components/modal/fullscreen-modal/), [ModalLayer](https://paragon-openedx.netlify.app/components/modal/modal-layer/) nor [ModalDialog](https://paragon-openedx.netlify.app/components/modal/modal-dialog/).

The UI has a modal-like overlay without a backdrop that anchors just over the Course Home tabs and extends to the bottom of the page.

I've managed to create some hooks to solve the UI. Maybe this can be revisited and replaced by some of Paragon's previously mentioned components in the future.

The overlay has some placeholder text to test the inner scrolling. When the modal pops up, it scrolls to the top of the page and freeze the `<body>` scrolling.

## Notes
I've added a prop to the slice `course-home/data/slice.js`. I don't see there's any unit tests on slices in the repo, maybe because they are small and simple or maybe they can be tested indirectly. I could add a `slice.test.js` for that extra prop if needed.

## Testing Instructions:

In order to see the search button and overlay:
- Follow the instructions to enable the waffle flags as indicated in [this PR](https://github.com/openedx/frontend-app-learning/pull/1199).
- You should be able to click the search icon to popup the overlay.

## Results:

You should be able to click and open the search modal when the waffle flag is enabled.
